File: poh20/php/8.php
#!/usr/bin/env php
<?php
$b=[448,56,7,292,146,73,273,84];
$a=intval(str_replace([".","#","\n"],[0,1,""],stream_get_contents(STDIN)),2);
echo count(array_filter($b,fn($e)=>($a&$e)==$e));

File: poh20/rust/8.rs
//usr/bin/env true; tmpfile=$(mktemp); rustc -o $tmpfile $0 && $tmpfile "$@"; rm $tmpfile; exit
use std::io::Read;
fn s()->String{let mut x=String::new();std::io::stdin().read_to_string(&mut x).ok();return x;}
fn main(){
let b:Vec<i32>=vec![448,56,7,292,146,73,273,84];
let x=s().replace("\n","").replace(".","0").replace("#","1");
let a=i32::from_str_radix(&x,2).unwrap();
println!("{}",b.iter().filter(|&&e|(a&e)==e).count());
}
